lib/widgets/validations/vtt_amount_input.dart

```diff
@@ -0,0 +1,106 @@
+};
+
+// Extend FormzInput and provide the input type and error type.
+class VttAmountInput extends FormzInput<String, String?> {
+  final int availableNanoWit;
+  final bool allowZero;
+  final bool allowValidation;
+  final int? weightedAmount;
+
+  // Call super.pure to represent an unmodified form input.
+  VttAmountInput.pure()
+      : availableNanoWit = 0,
+        allowZero = false,
+        weightedAmount = null,
+        allowValidation = false,
+        super.pure('');
+
+  // Call super.dirty to represent a modified form input.
+  VttAmountInput.dirty(
+      {required this.availableNanoWit,
+      value = '',
+      this.weightedAmount,
+      this.allowZero = false,
+      this.allowValidation = false})
+      : super.dirty(value);
+
+  int _witAmountToNanoWitNumber(String amount) {
+    try {
+      return num.parse(amount != '' ? amount : '0')
+          .standardizeWitUnits(
+              inputUnit: WitUnit.Wit, outputUnit: WitUnit.nanoWit)
+          .toBigInt()
+          .toInt();
+    } catch (e) {
+      print('Error parsing number $e');
+      return 0;
+    }
+  }
+
+  bool _notEnoughFunds({bool avoidWeightedAmountCheck = false}) {
+    int balance = this.availableNanoWit;
+    int nanoWitAmount;
+    if (!avoidWeightedAmountCheck) {
+      nanoWitAmount = this.weightedAmount ?? _witAmountToNanoWitNumber(value);
+    } else {
+      nanoWitAmount = _witAmountToNanoWitNumber(value);
+    }
+    return balance < nanoWitAmount;
+  }
+
+  // Override validator to handle validating a given input value.
+  @override
+  String? validator(String value, {bool avoidWeightedAmountCheck = false}) {
+    final validationUtils = ValidationUtils(errorMap: errorMap);
+    if (this.allowValidation) {
+      if (value.isEmpty)
+        return validationUtils.getErrorText(AmountInputError.empty);
+      if (RegExp(r'[a-zA-Z]').hasMatch(value))
+        return validationUtils.getErrorText(AmountInputError.invalid);
+      if (value.contains('.')) {
+        if (value.split('.').length != 2)
+          return validationUtils.getErrorText(AmountInputError.invalid);
+        if (value.split('.')[1].isEmpty)
+          return validationUtils.getErrorText(AmountInputError.invalid);
+        if (!RegExp(r'^[0-9]+(\.[0-9]{1,9})?$').hasMatch(value))
+          return validationUtils.getErrorText(AmountInputError.decimals);
+      }
+      if (!RegExp(r'^[0-9]{1,10}(\.[0-9]+)?$').hasMatch(value))
+        return validationUtils.getErrorText(AmountInputError.tooBig);
+      if (_notEnoughFunds(avoidWeightedAmountCheck: avoidWeightedAmountCheck))
+        return validationUtils.getErrorText(AmountInputError.notEnough);
+      if (value.isNotEmpty && !this.allowZero) {
+        try {
+          if (num.parse(value) == 0)
+            return validationUtils.getErrorText(AmountInputError.zero);
+        } catch (e) {
+          return validationUtils.getErrorText(AmountInputError.invalid);
+        }
+      }
+    }
+    return null;
+  }
```

Comment:
This is a bit hard to read. Could you remove one indentation by changing the first if condition and also adding some comments on each validation block explaining what you are checking in each?

```suggestion
  String? validator(String value, {bool avoidWeightedAmountCheck = false}) {
    final validationUtils = ValidationUtils(errorMap: errorMap);

    if (!this.allowValidation) {
      return null;
    }
    if (value.isEmpty)
      return validationUtils.getErrorText(AmountInputError.empty);
    if (RegExp(r'[a-zA-Z]').hasMatch(value))
      return validationUtils.getErrorText(AmountInputError.invalid);
    if (value.contains('.')) {
      if (value.split('.').length != 2)
        return validationUtils.getErrorText(AmountInputError.invalid);
      if (value.split('.')[1].isEmpty)
        return validationUtils.getErrorText(AmountInputError.invalid);
      if (!RegExp(r'^[0-9]+(\.[0-9]{1,9})?$').hasMatch(value))
        return validationUtils.getErrorText(AmountInputError.decimals);
    }
    if (!RegExp(r'^[0-9]{1,10}(\.[0-9]+)?$').hasMatch(value))
      return validationUtils.getErrorText(AmountInputError.tooBig);
    if (_notEnoughFunds(avoidWeightedAmountCheck: avoidWeightedAmountCheck))
      return validationUtils.getErrorText(AmountInputError.notEnough);
    if (value.isNotEmpty && !this.allowZero) {
      try {
        if (num.parse(value) == 0)
          return validationUtils.getErrorText(AmountInputError.zero);
      } catch (e) {
        return validationUtils.getErrorText(AmountInputError.invalid);
      }
    }
  }
```